### DISCUSSION TITLE
NodeJS 17.4.0 feature: node:readline/promises

### DISCUSSION THREAD

**Discussion by maka-io** [TypeStrong/ts-node#1638] [General]
I'm getting an error:
`TS2307: Cannot find module 'node:readline/promises' or its corresponding type declarations.`

When trying to use:
`import * as readline from 'node:readline/promises';`

From node version 17.4.0:
https://nodejs.org/api/readline.html#readline

Any idea why this feature would be having a problem?

**Comment by cspotcode** [comment]
Since this is a typechecking error, it is coming from TypeScript, not ts-node, which should narrow down your investigation.

node:readline/promises is declared in @types/node, right?  I would check your version of @types/node and then check the @types/node declarations to see if there are any unexpected surprises there.

https://typestrong.org/ts-node/docs/troubleshooting#understanding-errors
https://unpkg.com/@types/node <-- this'll show the latest version, but you should check which version you are using

**Reply by maka-io** [reply]
Thanks @cspotcode ! I'll take a look over there.  Much appreciated 🙏

**Reply by kyratzakos** [reply]
Thanks @cspotcode. For anyone having this issue in the future, try upgrading the @types/node or uninstall and install it again.
I was using version 18.0 and it upgraded to 20.11

**Comment by swalker326** [comment]
Was an answer to this issue found?

**Reply by dhirajchhetri** [reply]
did it got resolved

**Comment by noomly** [comment]
The issue here is that the promise-based api hasn't yet been merged in `@types/node`. Follow https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61115 for more information, it looks like it's already implemented and waiting for a code review.
